client/lib/rpc.rs

```diff
@@ -0,0 +1,329 @@
+    pub fn list_deploys(
+        self,
+        node_address: String,
+        maybe_block_hash: Option<BlockHash>,
+    ) -> Result<ListDeploysResult> {
+        let response_value = match maybe_block_hash {
+            Some(block_hash) => {
+                let params = GetBlockParams { block_hash };
+                ListDeploys::request_with_map_params(
+                    self.verbose,
+                    &node_address,
+                    self.rpc_id,
+                    params,
+                )?
+            }
+            None => ListDeploys::request(self.verbose, &node_address, self.rpc_id)?,
+        };
+        let get_block_result: GetBlockResult = serde_json::from_value(response_value)?;
+        let result = ListDeploysResult::from(get_block_result);
+        Ok(result)
+    }
+
+    /// Get a block from the node.
+    pub fn get_block(
+        self,
+        node_address: &str,
+        maybe_block_hash: Option<BlockHash>,
+    ) -> Result<GetBlockResult> {
+        let response_value = match maybe_block_hash {
+            Some(block_hash) => {
+                let params = GetBlockParams { block_hash };
+                GetBlock::request_with_map_params(self.verbose, &node_address, self.rpc_id, params)?
+            }
+            None => GetBlock::request(self.verbose, &node_address, self.rpc_id)?,
+        };
+        Ok(serde_json::from_value::<GetBlockResult>(response_value)?)
+    }
+}
+
+/// General purpose client trait for making requests to casper-node's HTTP endpoints.
+pub(crate) trait RpcClient {
+    const RPC_METHOD: &'static str;
+
+    /// Call a casper-node JSON-RPC endpoint.
+    fn request(verbose: bool, node_address: &str, rpc_id: u32) -> Result<Value> {
+        executor::block_on(async {
+            request(
+                verbose,
+                node_address,
+                rpc_id,
+                Self::RPC_METHOD,
+                Params::None(()),
+            )
+            .await
+        })
+    }
+
+    /// Call a casper node JSON-RPC endpoint with parameters.
+    fn request_with_map_params<T: IntoJsonMap>(
+        verbose: bool,
+        node_address: &str,
+        rpc_id: u32,
+        params: T,
+    ) -> Result<Value> {
+        executor::block_on(async {
+            request(
+                verbose,
+                node_address,
+                rpc_id,
+                Self::RPC_METHOD,
+                Params::from(params.into_json_map()),
+            )
+            .await
+        })
+    }
+}
+
+// TODO - If/when https://github.com/AtsukiTak/warp-json-rpc/pull/1 is merged and published,
+//        change `rpc_id` to a `jsonrpc_lite::Id`.
+async fn request(
+    verbose: bool,
+    node_address: &str,
+    rpc_id: u32,
+    method: &str,
+    params: Params,
+) -> Result<Value> {
+    let url = format!("{}/{}", node_address, RPC_API_PATH);
+    let rpc_req = JsonRpc::request_with_params(rpc_id as i64, method, params);
```

Comment:
We should be printing this if `verbose` (see https://github.com/CasperLabs/casper-node/blob/master/client/src/rpc.rs#L64-L70)